extensions/app-configuration/deployment/src/main/java/io/quarkiverse/azure/app/configuration/deployment/AzureAppConfigurationProcessor.java

```diff
@@ -0,0 +1,53 @@
+package io.quarkiverse.azure.app.configuration.deployment;
+
+import java.util.List;
+
+import com.azure.data.appconfiguration.implementation.ConfigurationSettingPage;
+import com.azure.data.appconfiguration.models.ConfigurationSetting;
+
+import io.quarkiverse.azure.app.configuration.AzureAppConfigurationConfigBuilder;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.builditem.ExtensionSslNativeSupportBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.RunTimeConfigBuilderBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.NativeImageProxyDefinitionBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.RuntimeInitializedClassBuildItem;
+
+public class AzureAppConfigurationProcessor {
+    @BuildStep
+    public void feature(BuildProducer<FeatureBuildItem> feature) {
+        feature.produce(new FeatureBuildItem("azure-app-configuration"));
+    }
+
+    @BuildStep
+    public void enableSsl(BuildProducer<ExtensionSslNativeSupportBuildItem> extensionSslNativeSupport) {
+        extensionSslNativeSupport.produce(new ExtensionSslNativeSupportBuildItem("azure-app-configuration"));
+    }
+
+    @BuildStep
+    public void azureAppConfigurationConfigFactory(BuildProducer<RunTimeConfigBuilderBuildItem> runTimeConfigBuilder) {
+        runTimeConfigBuilder.produce(new RunTimeConfigBuilderBuildItem(AzureAppConfigurationConfigBuilder.class.getName()));
+    }
+
+    @BuildStep
+    void nativeImage(
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitializedClasses,
+            BuildProducer<NativeImageProxyDefinitionBuildItem> nativeImageProxyDefinitions,
+            BuildProducer<ReflectiveClassBuildItem> reflectiveClasses) {
+
+        runtimeInitializedClasses.produce(new RuntimeInitializedClassBuildItem("io.netty.util.concurrent.GlobalEventExecutor"));
+        runtimeInitializedClasses.produce(new RuntimeInitializedClassBuildItem(
+                "com.azure.core.http.vertx.VertxAsyncHttpClientProvider$GlobalVertxHttpClient"));
+        runtimeInitializedClasses.produce(
+                new RuntimeInitializedClassBuildItem("com.azure.core.http.vertx.VertxAsyncHttpClientBuilder$DefaultVertx"));
```

Comment:
We cannot rely on the existent extension for configuration because we need a Vert.x instance before is created by Quarkus, since the configuration is executed before everything else.

We could move some of these to the core module.

Reply:
> We could move some of these to the core module.

Now I see there is a dependency on `<artifactId>quarkus-azure-core</artifactId>` I think it was not there when I looked last time. That being the case now, could you please try to remove all these build items to see whether they are needed? I guess most of them might already be present in `quarkus-azure-core` or its transitives.

Reply:
Yes, I added the dependency and moved the required build items to core. The remaining ones are specific to the `azure-data-appconfiguration` dependency, which is only used by this extension.